### PR TITLE
[Naitei] ft:Edit_Info_User

### PR DESCRIPTION
**Pull request by ThaiKhacDuoc**
## Related Tickets
- ticket redmine

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)

image

### CODE REVIEW

**Comment by ThaiKhacDuoc** [comment]
ready

**Comment by ThaiKhacDuoc** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class UsersController < ApplicationController
+  include SessionsHelper
+  before_action :logged_in_user, only: %i(edit update)
+  before_action :find_user, only: %i(edit update)
+  before_action :correct_user, only: %i(edit update)
```
Comment (ghost) [inline app/controllers/users_controller.rb]: gộp lại đi em

File: app/views/users/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  </div>
+  <div class="auth-form__group">
+    <%= f.password_field :password, placeholder: t("sign_up.password"), class: "formdk__taikhoan" %>
+  </div>
+  <div class="auth-form__group">
+    <%= f.password_field :password_confirmation, placeholder: t("sign_up.password_confirmation"), class: "formdk__taikhoan" %>
+  </div>
+  <div class="auth-form__group">
+    <%= f.label :gender, t("sign_up.gender") %>
+    <%= f.select :gender, User.genders.keys.map { |gender| [t("sign_up.genders.#{gender}"), gender] }, {}, class: "formdk__gioitinh" %>
```
Comment (ghost) [inline app/views/users/_form.html.erb]: `User.genders.keys.map { |gender| [t("sign_up.genders.#{gender}"), gender] }` đưa cái này vào helper nhé

File: app/views/users/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  <div class="nav-link">
+    <div class="nav-link-text">
+      <i class="fa-solid fa-house"></i>
+      <i class="right fa-solid fa-chevron-right"></i>
+      <%= link_to t("profile.personal_info"), "#", class: 'nav-link--selected' %>
+    </div>
+  </div>
+  <div class="body">
+    <div class="body__menu">
+      <div class="title"><%= t("profile.personal_info") %></div>
```
Comment (ghost) [inline app/views/users/edit.html.erb]: bỏ () ở những chỗ có thể

File: app/views/users/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          <% end %>
+        </div>
+        <div class="list__item">
+          <%= link_to '#' do %>
+            <i class="list__icon fa-solid fa-wallet"></i>
+            <span><%= t("profile.order_history") %></span>
+          <% end %>
+        </div>
+        <div class="list__item">
+          <%= link_to '#' do %>
```
Comment (ghost) [inline app/views/users/edit.html.erb]: Dùng "

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ThaiKhacDuoc** [comment]
ready

**Comment by ThaiKhacDuoc** [comment]
ready

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -15,8 +19,43 @@ def create
...
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    redirect_to root_path unless @user
+  end
+
+  def logged_in_user
+    return if logged_in?
+
+    flash[:danger] = t("please_log_in")
```
Comment (ghost) [inline app/controllers/users_controller.rb]: Bỏ bớt () đi em

File: app/views/users/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<%= form_for @user, class: "js-RegisterModal" do |f| %>
+  <%= render "shared/error_messages", object: f.object%>
```
Comment (ghost) [inline app/views/users/_form.html.erb]:
```suggestion
  <%= render "shared/error_messages", object: f.object %>
```

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ThaiKhacDuoc** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**